### PR TITLE
fix(meth): emit -R read group as @RG header in --meth mode

### PR DESCRIPTION
**Pull request by tfenne**
## Problem

`bwa-mem3 mem --meth` stamps every record with an `RG:Z` tag (from the global `bwa_rg_id`, set by `-R`) but its BAM writer never received the assembled `hdr_line`, so **no `@RG` header line was written**. The output BAM declares no read group yet every record references one — malformed per the SAM spec and rejected by strict validators (Picard, noodles). The default (non-meth) path threads `hdr_line` into `bam_writer_open` and was always correct; only `--meth` diverged.

### Reproduction (before fix)

```
$ bwa-mem3 mem --meth -R $'@RG\tID:rg\tSM:s' ref.fa reads.fq | samtools view -H -
@HD     VN:1.6  SO:unsorted
@SQ     SN:chr1 LN:400
@PG     ...                 # <- no @RG line
$ bwa-mem3 mem --meth -R $'@RG\tID:rg\tSM:s' ref.fa reads.fq | samtools view - | grep -o RG:Z:rg
RG:Z:rg                     # <- but records carry RG:Z:rg
```

The non-meth path (`mem -R ...`) emits the `@RG` line correctly; only `--meth` dropped it.

## Fix

Thread `hdr_line` into `meth_bam_writer_open` and emit it **after the `@SQ` block and before the `@PG` lines**, mirroring `bam_writer_open`'s precedence (user header > index > default). Also suppress the default `@HD` when the user's `-H` supplies one, since htslib's `sam_hdr_add_lines` does not de-dup `@HD`.

The consolidated `@SQ` remains authoritative for `--meth`: a user `-H` that re-supplies an already-emitted contig now fails the open loudly (duplicate-SN error) rather than emitting duplicate references — out of scope to merge here, and a loud failure is preferable to silent corruption.

### After fix

```
@HD     VN:1.6  SO:unsorted
@SQ     SN:chr1 LN:400
@RG     ID:rg   SM:s          # <- now present, between @SQ and @PG
@PG     ID:bwa-mem3 ...
@PG     ID:bwa-mem3-meth ...
```

## Test

Adds `test/meth_rg_header_test.sh` (wired into `run_unit_tests.sh` next to `pg_cl_escape_test`). It runs `mem` and `mem --meth` with the same `-R` and asserts the SAM invariant that **every `RG:Z` value is declared by an `@RG ID` in the header**. The first commit adds the test (RED on `--meth`); the second commit applies the fix (GREEN). The test drives the stable `-R` CLI rather than the C signature, so it survives refactors. The `--meth` half skips with a warning when `samtools` is absent.

Verified locally: full `test/run_unit_tests.sh` passes, including the new test, and an `-H` `@HD`/`@CO` + `-R` edge case produces exactly one `@HD` with correct ordering.

## Summary by CodeRabbit

* **New Features**
  * User-supplied SAM header text (via `-R`/`-H`) is now properly preserved in `--meth` BAM output, with correct ordering.
  * Read group (`-R `@RG` ...`) headers are applied consistently to all alignment records in both SAM and `--meth` BAM paths.
  * Prevents duplicate `@HD` entries when a header already provides one.

* **Tests**
  * Added `meth_rg_header_test` regression coverage for `@RG`/`RG:Z` consistency across SAM and `--meth` BAM, integrated into the unit test runner.

### CODE REVIEW

File: test/meth_rg_header_test.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    echo "SKIP: samtools not found; --meth half not checked" >&2
+    echo "PASS: @RG header emitted in non-meth mode (--meth skipped)"
+    exit 0
+fi
+
+# --meth requires a c2t-converted index built with `index --meth`.
+if [[ ! -s "$ref.bwameth.c2t.bwt.2bit.64" ]]; then
+    "$bin" index --meth "$ref" >/dev/null 2>&1 \
+        || { echo "FAIL: bwa-mem3 index --meth on phix.fa failed" >&2; exit 1; }
+fi
```
[the remark this answers is not in the document]
Comment (tfenne) [reply test/meth_rg_header_test.sh]: Good catch — fixed in 80e62d5. The `--meth` precheck now verifies the converted FASTA plus all five c2t FMI artifacts (`.bwameth.c2t`, `.bwt.2bit.64`, `.0123`, `.amb`, `.ann`, `.pac`), mirroring the non-meth block above, so a partial index from an interrupted run is rebuilt instead of triggering a spurious failure. Verified by deleting one artifact and confirming the test rebuilds and passes.

**Comment by nh13** [comment]
Thanks @tfenne. I pushed one addition (`e878d68`): a `--meth` `@HD` de-dup test pinning the `hdr_text_has_type()` guard, plus comments noting we intentionally don't forward the index `.hdr/.dict` sidecar in `--meth` (it's the `.bwameth.c2t` index). Filed #138 for the real `@SQ` `M5`/`UR` enrichment follow-on. Will merge once CI is green.
